### ISSUE TITLE
installation issues on WSL2 with Windows 10

### ISSUE THREAD

**Issue by alexlib** [yangyushi/FishPy#14]
Not sure if it's a general or a specific computer problem. Here's the log of the compilation failure

```
(fishpy) alex@DESKTOP-OV3L1FT:/mnt/c/Users/alex/Downloads/fishpy/fishpy$ ./build.sh
rm -f *.so
rm -f *.o
rm -f *.a
clang++ -Wall -I/usr/local/include/eigen3 -std=c++11 -O3 -I/mnt/c/Users/alex/Downloads/fishpy/ILOG/cplex/include -I/mnt/c/Users/alex/Downloads/fishpy/ILOG/concert/include -DIL_STD -I. -fPIC -c greta.cpp -o greta.o
ar rcs libgreta.a greta.o
rm -f greta.o
clang++ -Wall -I/usr/local/include/eigen3 -std=c++11 -O3 -I/mnt/c/Users/alex/Downloads/fishpy/ILOG/cplex/include -I/mnt/c/Users/alex/Downloads/fishpy/ILOG/concert/include -DIL_STD -I. -fPIC -c temporal.cpp -o temporal.o
ar rcs libtemporal.a temporal.o
rm -f temporal.o
clang++ -Wall -I/usr/local/include/eigen3 -std=c++11 -O3 -I/mnt/c/Users/alex/Downloads/fishpy/ILOG/cplex/include -I/mnt/c/Users/alex/Downloads/fishpy/ILOG/concert/include -DIL_STD -I. -fPIC -c stereo.cpp -o stereo.o
ar rcs libstereo.a stereo.o
rm -f stereo.o
clang++ -Wall -I/usr/local/include/eigen3 -std=c++11 -O3 -I/mnt/c/Users/alex/Downloads/fishpy/ILOG/cplex/include -I/mnt/c/Users/alex/Downloads/fishpy/ILOG/concert/include -DIL_STD -I. -fPIC -I/home/alex/anaconda3/envs/fishpy/include/python3.9 -I/home/alex/anaconda3/envs/fishpy/lib/python3.9/site-packages/pybind11/include -L/home/alex/anaconda3/envs/fishpy/lib/python3.9/config-3.9-x86_64-linux-gnu -L/home/alex/anaconda3/envs/fishpy/lib  -lcrypt -lpthread -ldl  -lutil -lrt -lm -lm  -shared cgreta.cpp libgreta.a libtemporal.a libstereo.a -L/mnt/c/Users/alex/Downloads/fishpy/ILOG/cplex/lib/x86-64_linux/static_pic -L//mnt/c/Users/alex/Downloads/fishpy/ILOG/concert/lib/x86-64_linux/static_pic -lilocplex -lconcert -lcplex -lm -lpthread -ldl -o cgreta.cpython-39-x86_64-linux-gnu.so
clang++ -Wall -I/usr/local/include/eigen3 -std=c++11 -O3 -I/mnt/c/Users/alex/Downloads/fishpy/ILOG/cplex/include -I/mnt/c/Users/alex/Downloads/fishpy/ILOG/concert/include -DIL_STD -I. -fPIC -I/home/alex/anaconda3/envs/fishpy/include/python3.9 -I/home/alex/anaconda3/envs/fishpy/lib/python3.9/site-packages/pybind11/include -L/home/alex/anaconda3/envs/fishpy/lib/python3.9/config-3.9-x86_64-linux-gnu -L/home/alex/anaconda3/envs/fishpy/lib  -lcrypt -lpthread -ldl  -lutil -lrt -lm -lm  -shared cstereo.cpp libstereo.a -L/mnt/c/Users/alex/Downloads/fishpy/ILOG/cplex/lib/x86-64_linux/static_pic -L//mnt/c/Users/alex/Downloads/fishpy/ILOG/concert/lib/x86-64_linux/static_pic -lilocplex -lconcert -lcplex -lm -lpthread -ldl -o cstereo.cpython-39-x86_64-linux-gnu.so
clang++ -Wall -I/usr/local/include/eigen3 -std=c++11 -O3 -I/mnt/c/Users/alex/Downloads/fishpy/ILOG/cplex/include -I/mnt/c/Users/alex/Downloads/fishpy/ILOG/concert/include -DIL_STD -I. -fPIC -I/home/alex/anaconda3/envs/fishpy/include/python3.9 -I/home/alex/anaconda3/envs/fishpy/lib/python3.9/site-packages/pybind11/include -L/home/alex/anaconda3/envs/fishpy/lib/python3.9/config-3.9-x86_64-linux-gnu -L/home/alex/anaconda3/envs/fishpy/lib  -lcrypt -lpthread -ldl  -lutil -lrt -lm -lm  -shared stereo.cpp cray_trace.cpp -L/mnt/c/Users/alex/Downloads/fishpy/ILOG/cplex/lib/x86-64_linux/static_pic -L//mnt/c/Users/alex/Downloads/fishpy/ILOG/concert/lib/x86-64_linux/static_pic -lilocplex -lconcert -lcplex -lm -lpthread -ldl -o cray_trace.cpython-39-x86_64-linux-gnu.so
clang++ -Wall -I/usr/local/include/eigen3 -std=c++11 -O3 -I/mnt/c/Users/alex/Downloads/fishpy/ILOG/cplex/include -I/mnt/c/Users/alex/Downloads/fishpy/ILOG/concert/include -DIL_STD -I. -fPIC -I/home/alex/anaconda3/envs/fishpy/include/python3.9 -I/home/alex/anaconda3/envs/fishpy/lib/python3.9/site-packages/pybind11/include -L/home/alex/anaconda3/envs/fishpy/lib/python3.9/config-3.9-x86_64-linux-gnu -L/home/alex/anaconda3/envs/fishpy/lib  -lcrypt -lpthread -ldl  -lutil -lrt -lm -lm  -shared ctemporal.cpp libtemporal.a -L/mnt/c/Users/alex/Downloads/fishpy/ILOG/cplex/lib/x86-64_linux/static_pic -L//mnt/c/Users/alex/Downloads/fishpy/ILOG/concert/lib/x86-64_linux/static_pic -lilocplex -lconcert -lcplex -lm -lpthread -ldl -o ctemporal.cpython-39-x86_64-linux-gnu.so
clang++ -Wall -I/usr/local/include/eigen3 -std=c++11 -O3 -I/mnt/c/Users/alex/Downloads/fishpy/ILOG/cplex/include -I/mnt/c/Users/alex/Downloads/fishpy/ILOG/concert/include -DIL_STD -I. -fPIC -I/home/alex/anaconda3/envs/fishpy/include/python3.9 -I/home/alex/anaconda3/envs/fishpy/lib/python3.9/site-packages/pybind11/include -L/home/alex/anaconda3/envs/fishpy/lib/python3.9/config-3.9-x86_64-linux-gnu -L/home/alex/anaconda3/envs/fishpy/lib  -lcrypt -lpthread -ldl  -lutil -lrt -lm -lm  -shared cutility.cpp -L/mnt/c/Users/alex/Downloads/fishpy/ILOG/cplex/lib/x86-64_linux/static_pic -L//mnt/c/Users/alex/Downloads/fishpy/ILOG/concert/lib/x86-64_linux/static_pic -lilocplex -lconcert -lcplex -lm -lpthread -ldl -o cutility.cpython-39-x86_64-linux-gnu.so
make: *** No rule to make target 'clean'.  Stop.
clang++ -Ofast -Wall -I. -I/usr/local/include/voro++ -std=c++14 -fPIC -shared -mtune=native -march=native `python3 -m pybind11 --includes` `python3-config --ldflags` voro.cpp -L/usr/local/lib -lvoro++ -o voro.cpython-39-x86_64-linux-gnu.so
/usr/bin/ld: /usr/local/lib/libvoro++.a(cell.o): relocation R_X86_64_PC32 against symbol `_ZTVN4voro16voronoicell_baseE' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:9: voro.cpython-39-x86_64-linux-gnu.so] Error 1
make: *** No rule to make target 'clean'.  Stop.
clang++ -L/usr/local/lib -Wall -I/usr/local/include/eigen3 -std=c++14 -fPIC -shared -O3 -I/home/alex/anaconda3/envs/fishpy/include/python3.9 -I/home/alex/anaconda3/envs/fishpy/lib/python3.9/site-packages/pybind11/include -L/home/alex/anaconda3/envs/fishpy/lib/python3.9/config-3.9-x86_64-linux-gnu -L/home/alex/anaconda3/envs/fishpy/lib  -lcrypt -lpthread -ldl  -lutil -lrt -lm -lm  nrook.cpp -o nrook.cpython-39-x86_64-linux-gnu.so

```

**Comment by yangyushi** [comment]
Hi @alexlib, I think the issue is fixed with 10b180cc3144b9c5b2416905fd300f793d8456f1, after which the `voro++` package should not cause any error for FishPy.

I am still working on the compiling process, to make it easier for user to compile.

I am not closing the issue for now, because I guess the package is still "hard to build". Hopefully it gets better soon.

**Comment by alexlib** [comment]
this is resolved. thanks

i still encounter problems. next time :)
